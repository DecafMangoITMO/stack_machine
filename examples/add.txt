section .data:
      a: 1
      b: 1
  section .text:
      lit a
      push
      lit b
      push
      add
      halt