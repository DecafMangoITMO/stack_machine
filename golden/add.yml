in_source: |
  section .data:
      a: 1
      b: 1
  section .text:
      lit a
      push
      lit b
      push
      add
      halt
in_stdin: |-
  []
out_code: |-
  00000000000000000000000000000110
  00000000000000000000000000000000
  00000000000000000000000000000000
  00000000000000000000000000000000
  00000000000000000000000000000001
  00000000000000000000000000000001
  10001000000000000000000000000000
  00000000000000000000000000000100
  10010000000000000000000000000000
  10001000000000000000000000000000
  00000000000000000000000000000101
  10010000000000000000000000000000
  00001000000000000000000000000000
  11000000000000000000000000000000
out_commented_code: |-
  0  00000000000000000000000000000110
  1  00000000000000000000000000000000
  2  00000000000000000000000000000000
  3  00000000000000000000000000000000
  4  00000000000000000000000000000001 1
  5  00000000000000000000000000000001 1
  6  10001000000000000000000000000000 lit
  7  00000000000000000000000000000100 4
  8  10010000000000000000000000000000 push
  9  10001000000000000000000000000000 lit
  10 00000000000000000000000000000101 5
  11 10010000000000000000000000000000 push
  12 00001000000000000000000000000000 add
  13 11000000000000000000000000000000 halt
out_stdout: |
  source LoC: 10 code instr: 14
  ============================================================


  instr_counter: 5 ticks: 30
out_log: |
  DEBUG   machine:execute_lit   TICK: 6          PC: 8          TODS1: 4          TODS2: 0          TOAS: 0          Z_FLAG: 0 	lit 4
  	   DATA_STACK: [4]
  	   ADDRESS_STACK: []
  DEBUG   machine:execute_push  TICK: 12         PC: 9          TODS1: 1          TODS2: 0          TOAS: 9          Z_FLAG: 0 	push
  	   DATA_STACK: [1]
  	   ADDRESS_STACK: []
  DEBUG   machine:execute_lit   TICK: 16         PC: 11         TODS1: 5          TODS2: 0          TOAS: 9          Z_FLAG: 0 	lit 5
  	   DATA_STACK: [1, 5]
  	   ADDRESS_STACK: []
  DEBUG   machine:execute_push  TICK: 22         PC: 12         TODS1: 1          TODS2: 0          TOAS: 12         Z_FLAG: 0 	push
  	   DATA_STACK: [1, 1]
  	   ADDRESS_STACK: []
  DEBUG   machine:execute_binary_math_instruction TICK: 28         PC: 13         TODS1: 2          TODS2: 1          TOAS: 12         Z_FLAG: 0 	add
  	   DATA_STACK: [2]
  	   ADDRESS_STACK: []
  DEBUG   machine:execute_halt  TICK: 30         PC: 14         TODS1: 2          TODS2: 1          TOAS: 12         Z_FLAG: 0 	halt
  	   DATA_STACK: [2]
  	   ADDRESS_STACK: []
